auto_gptq/utils/marlin_utils.py

```diff
@@ -142,7 +142,9 @@ def convert_to_marlin(model, model_quantlinear, quantization_config, repack: boo
         if not isinstance(module, model_quantlinear):
             continue
 
-        if module.bias is not None and torch.count_nonzero(module.bias) > 0:
+        # We could use `torch.count_nonzero(module.bias) > 0` here to discard zero bias, but this has issues when loading weights
```

Comment:
My opinion that instead of making such guards, it's better to not always create a bias, it should depend on whether the module (which will be replaced by QuantLinear) has bias or not.

But here (and on line 124) it is hard-coded to be always created:
https://github.com/AutoGPTQ/AutoGPTQ/blob/10cd170ac8b295e6f64422a655a5a44f07ff0286/auto_gptq/modeling/_utils.py#L113

Reply:
@Andrei-Aksionov what I meant is that checkpoints as https://huggingface.co/TheBloke/Llama-2-7B-Chat-GPTQ?show_tensors=true have `bias` tensors, although they are all zeros.

Let's fix that in an other PR https://github.com/AutoGPTQ/AutoGPTQ/issues/555

Reply:
Of course :) That's because the current implementation creates bias values no matter what. Since in all QuantLinear classes a bias is initialized with zeros, that might explain why this checkpoint has it.

So my suggestion is to fix this hard-coded bias creation (in a separate PR) and have a guard in the code that loads a checkpoint.